cmd/gcppromd/main.go

```diff
@@ -4,159 +4,63 @@ import (
 	"context"
 	"flag"
 	"net/http"
+	"os"
 	"strings"
 
 	"github.com/messagebird/gcppromd"
 
 	"bytes"
 	"encoding/json"
-	"github.com/go-chi/chi"
-	"github.com/go-chi/chi/middleware"
-	"github.com/go-chi/render"
 	log "github.com/sirupsen/logrus"
 	"io/ioutil"
 	"time"
 )
 
-type ctxKey int
-
 const (
-	gcedsCtxKey ctxKey = iota
+	projectSeparator   = ","
+	defaultGCEPoolSize = 20
 )
 
-const (
-	projectSeparator = ","
-	gceWorkers       = 20
+var (
+	flisten    = flag.String("listen", ":8080", "HTTP listen address")
+	fdaemon    = flag.Bool("daemon", false, "run the application as a daemon that periodically produces a target file with a json in Prometheus file_sd format. Disables web-mode")
+	fouput     = flag.String("outputPath", "/etc/prom_sd/targets.json", "A path to the output file with targets")
+	fdiscovery = flag.Int64("frequency", 300, "discovery frequency in seconds")
+	fprojects  = flag.String("projects", "", "comma-separated projects IDs")
 )
 
 func main() {
-	var (
-		listenAddr = flag.String("listen", ":8080", "HTTP listen address")
-		daemonMode = flag.Bool("daemon", false, "run the application as a daemon that periodically "+
-			"produses a targets file with a json in Prometheus file_sd format. Disables web-mode")
-		outputFile = flag.String("outputPath", "/etc/prom_sd/targets.json", "A path to the output"+
-			"file with targets")
-		discoveryFreq = flag.Int64("frequency", 300, "discovery frequency in seconds")
-		projects      = flag.String("projects", "", "comma-separated projects IDs")
-	)
 	flag.Parse()
 
-	gceds, err := gcppromd.NewGCEDiscoveryPool(context.TODO(), gceWorkers)
-	if err != nil {
-		log.WithFields(log.Fields{
-			"err":    err,
-			"listen": *listenAddr,
-		}).Fatal("Cannot initialise GCE discovery")
-	}
+	gceds, err := gcppromd.NewGCEDiscoveryPool(context.TODO(), defaultGCEPoolSize)
```

Comment:
Why use todo? Create a context with cancel, and use it for graceful shutdown. In the go funcs `NewGCEDiscoveryPool` starts you have a case for `ctx.Done()` but you don't do anything anywhere with it?
Listen for sigint, sigterm a bit down and cancel it.